crates/ide/src/hover/render.rs

```diff
@@ -577,17 +577,17 @@ where
 fn label_and_layout_info_and_docs<D, E, V>(
     db: &RootDatabase,
     def: D,
+    config: &HoverConfig,
     value_extractor: E,
 ) -> (String, Option<hir::Documentation>)
 where
     D: HasAttrs + HirDisplay,
     E: Fn(&D) -> Option<V>,
     V: Display,
 {
-    let label = if let Some(value) = value_extractor(&def) {
-        format!("{} // {value}", def.display(db))
-    } else {
-        def.display(db).to_string()
+    let label = match value_extractor(&def) {
+        Some(value) if config.memory_layout => format!("{} // {value}", def.display(db)),
```

Comment:
Unfortunately closure hover doesn't use `label_and_layout_info_and_docs`, but it shows some layout info which needs to be disabled.

Reply:
Thank you, I missed that closures use a different implementation.
I've added an additional test for closures.